### DISCUSSION TITLE
Delete only compose.yam and .env file

### DISCUSSION THREAD

**Discussion by Celer21** [louislam/dockge#405] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API

### 🔖 Feature description

Hi,
would it be possible to delete only compose.yml and .env file while deleting stack?
Other files and folders would not be deleted.

### ✔️ Solution

In my case, I would like to store data inside Stack folder.

For example i have this stack:
Stacks\uptime-kuma\                          -- do not delete
Stacks\uptime-kuma\compose.yaml  -- delete 
Stacks\uptime-kuma\.env                   -- delete
Stacks\uptime-kuma\data\db.sql       -- do not delete

### 📝 Additional Context

This coud be parameter in setting. 
I like keep data and compose file togeder in one folder / subfolder for easy beckup and management.

**Comment by louislam** [comment]
Alternatively, you can use the `Inactive` button or `docker-compose down`.

**Reply by trythatagain** [reply]
I see how this might be useful, but for me I'd rather use another folder that I call 'appdata' which holds all my containers data. It's saved my butt a number times :) when I've deleted containers. As I use OMV I have a global env file (which I believe is being worked on at this moment) and appdata is defined in there. You could also use a symlink if the path might be full UUID path.

**Comment by wsw70** [comment]
Do you mean that the "Delete" button removes **persistent** data (mounted directories or volumes")??

EDIT: I just tested and fortunately not, phhew.

**Reply by Celer21** [reply]
No, dockge only deletes whole stack folder.
So if you have persistent data in your stack folder, Dockge delete whole folder with your persistent data.

In my case, i want to have my comspose file and persistend data in one folder for example /docker/stacks/test
After delete my stack "Test", i lost persistent data.

![image](https://github.com/user-attachments/assets/9d862058-711c-4eaa-a1d6-01c4e0bfc0c2)
![image](https://github.com/user-attachments/assets/e19d1965-9cc5-49ce-9d3a-0d4c69a9b871)

If you have stack and data directory separate, it is not problem.
